### PR TITLE
Freemius uninstall

### PR DESCRIPTION
**Pull request by zbtirrell**
uninstall is firing and the reason is sent to Freemius.

see: 
https://central.tri.be/issues/138342 (100% of installs)
https://central.tri.be/issues/134066 (fix uninstall hook)

Note: the uninstall event is causing an email to be sent to the customer if the plugin is deleted soon after being installed. This is non-configurable and does not meet our quality standards, were reaching out to Freemius about this, but does not block progress on this work

### CODE REVIEW

**Comment by bordoni** [comment]
What might be required is the correct path being set inside of the `get_caller_main_file_and_type` from Freemius or inside of the option set in `store_id_slug_type_path_map`.

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -3019,6 +3019,15 @@ public static function deactivate( $network_deactivating ) {
 			add_action( 'shutdown', array( $deactivation, 'deactivate' ) );
 		}
 
+		/**
+		 * plugin uninstall callback
+		 * @see register_uninstall_hook()
+		 */
+		public static function uninstall() {
+			Tribe__Events__Integrations__Manager::instance()->load_freemius();
+			tribe( 'events.integrations.freemius' )->uninstall();
```
Comment (bordoni) [inline src/Tribe/Main.php]: This method doesn't exist in the Freemius class or ours.

Comment (zbtirrell) [reply src/Tribe/Main.php]: yeah, I wanted it to fatal, but it never gets to this point in the code :(

**Comment by zbtirrell** [comment]
updated PR with an adjusted approach. Edited description with current state of work

**Comment by zbtirrell** [comment]
This is now fully functional!

File: src/Tribe/Integrations/Freemius.php [hunk 2]
```diff
@@ -64,7 +64,10 @@ public function __construct() {
 			'tribe-help',
 		];
 
-		if ( ! in_array( $this->page, $valid_page ) && 'plugins.php' !== $pagenow ) {
+		if ( ! in_array( $this->page, $valid_page )
+			&& 'plugins.php' !== $pagenow
+			&& 'admin-ajax.php' !== $pagenow ) {
```
Comment (borkweb) [inline src/Tribe/Integrations/Freemius.php]:
```php
if (
	! in_array( $this->page, $valid_page )
	&& 'plugins.php' !== $pagenow
	&& 'admin-ajax.php' !== $pagenow
) {
```

File: src/Tribe/Integrations/Freemius.php [hunk 3]
```diff
@@ -64,7 +64,10 @@ public function __construct() {
...
+			echo 'page: ' . $pagenow;
```
Comment (borkweb) [inline src/Tribe/Integrations/Freemius.php]: Is this debug output?

**Comment by bordoni** [comment]
Waiting for tests to pass, and will merge.

**Comment by bordoni** [comment]
Both fails are unrelated. One is a false positive, the other snapshots need to be regenerated on Views V2. Will look into this a little further on a separate PR.

**Pull request merged**